src/main.rs

```diff
@@ -55,17 +118,405 @@ impl BFInterp {
+                        _ => todo!(),
+                    },
+                    _ => todo!(),
+                }
+            }
+            ConstantOperand(con_op) => match con_op.as_ref() {
+                Constant::Int { value, .. } => value.to_string(),
+                Constant::GetElementPtr(GetElementPtr { address, .. }) => match address.as_ref() {
+                    Constant::GlobalReference { name, .. } => {
+                        match self.gl_vars.get(name).unwrap() {
+                            Constant::Array { elements, .. } => {
+                                let mut string = String::new();
+                                for elem in elements.iter() {
+                                    match elem.as_ref() {
+                                        Constant::Int { value, .. } => {
+                                            if value != &0u64 {
+                                                string.push(*value as u8 as char)
+                                            }
+                                        }
+                                        _ => todo!(),
+                                    }
+                                }
+                                string
+                            }
+                            _ => todo!(),
+                        }
+                    }
+                    _ => todo!(),
+                },
+                _ => todo!(),
+            },
+            MetadataOperand => todo!(),
+        }
+    }
+
+    fn int_bin_operations(
+        &mut self,
+        op0: &Operand,
+        op1: &Operand,
+        dest: &name::Name,
+        operation_type: u8,
+    ) {
+        match op0 {
+            LocalOperand { name, ty } => match ty.as_ref() {
+                Type::IntegerType { bits } => match bits {
+                    8 => todo!(),
+                    16 => todo!(),
+                    32 => {
+                        let op0 = self.get_i32_op(self.vars.get(name).unwrap()); //int_vars
+                        let op1 = self.get_i32_op(op1);
+                        self.i32_bin_operation(op0, op1, dest, operation_type)
+                    }
+                    64 => {
+                        let op0 = self.get_i64_op(self.vars.get(name).unwrap());
+                        let op1 = self.get_i64_op(op1);
+                        self.i64_bin_operation(op0, op1, dest, operation_type)
+                    }
+                    128 => todo!(),
+                    _ => todo!(),
+                },
+                _ => todo!(),
+            },
+            ConstantOperand(con_op) => match con_op.as_ref() {
+                Constant::Int { value, .. } => {
+                    *value as i32;
+                }
+                Constant::Vector(_) => todo!(),
+                _ => todo!(),
+            },
+            MetadataOperand => todo!(),
+        }
+    }
+
+    fn get_i32_op(&self, op: &Operand) -> i32 {
```

Comment:
I think we can combine this and the next function into one by using generics: https://doc.rust-lang.org/book/ch10-01-syntax.html#generic-data-types

Reply:
I don't think this is possible because op is of type &Operand rather than T.

Reply:
I think Lukas means something like `fn get_32_op<T>(&self, op: &T) -> i32 { ... }`.

Reply:
But the whole point of the method is to get the value from the operand.

Reply:
I meant something like `fn get_int<T>(&self, op: &Operand) -> T`, where `T` can be `u32`, `u64`, etc.

Reply:
Can you have a look at the changes I've done? It's possible this is a worse way to implement this than before.